base_layer/core/src/mempool/sync_protocol/initializer.rs

```diff
@@ -103,9 +107,9 @@ impl ServiceInitializer for MempoolSyncInitializer {
                 }
                 log_mdc::extend(mdc.clone());
             }
-
-            MempoolSyncProtocol::new(config, notif_rx, connectivity_event_subscription, mempool)
-                .run()
+            let block_event_stream = base_node.get_block_event_stream();
+            MempoolSyncProtocol::new(config, notif_rx, connectivity_event_subscription, mempool, connectivity)
```

Comment:
I would get rid of the events receiver param and get it inside the run function now that we're passing in connectivity

```suggestion
            MempoolSyncProtocol::new(config, notif_rx, mempool, connectivity)
```